orttraining/orttraining/core/graph/gradient_builder.cc

```diff
@@ -711,12 +711,8 @@ IMPLEMENT_GRADIENT_BUILDER(GetDropoutGradient) {
 
 IMPLEMENT_GRADIENT_BUILDER(GetConvGradient) {
   std::vector<ArgDef> outputs;
-  for (int i = 0; i < 3; i++) {
-    if (IsGradientRequiredForSrcNodeInput(i)) {
-      outputs.push_back(GI(i));
-    } else {
-      outputs.push_back(ArgDef("", nullptr));
-    }
+  for (int i = 0; i < GetSrcNodeInputSize(); i++) {
```

Comment:
This was actually an intended optimization. ORT would only compute the gradient for inputs that requires the gradient. 
The ConvGrad schema should be updated instead, all three outputs, dW, dB, dX should be optional.

Reply:
It seems that as of a few months ago `dX` and `dB` now have the optional schema (I assume `dW` still needs it as well)
https://github.com/microsoft/onnxruntime/pull/5462

However the implementation of the kernel doesn't check if `dX` is present or not, so that needs to be updated
https://github.com/microsoft/onnxruntime/blob/master/orttraining/orttraining/training_ops/cpu/nn/conv_grad.cc

I'll update this PR within the next few weeks to add the `optional` tag to `dW` and update the kernel accordingly

Reply:
@SherlockNoMad I've updated the schema & kernel to mark `dW` as optional.
@georgen117 since `dX` is already marked optional, I think the error in #4762 should not occur using a latest build of ORT. This PR then only fixes the case where bias is not present, and the rarer case where the weights are not marked for training.

Reply:
Thanks for a lot for help fixing this. I signed off.